packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -26,7 +26,7 @@ @interface InAppPurchasePlugin ()
 // Callback channel to dart used for when a function from the payment queue delegate is triggered.
 @property(strong, nonatomic, readonly) FlutterMethodChannel *paymentQueueDelegateCallbackChannel;
 
-@property(strong, nonatomic, readonly) NSObject<FlutterTextureRegistry> *registry;
+@property(weak, nonatomic, readonly) NSObject<FlutterTextureRegistry> *registry;
```

Comment:
Looks like registry is unused in the plugin code and can be completely removed. "messenger" can also be removed with a little tweak in the code.

And it looks like registrar can be weak as the engine should keep a strong reference to the registrar and plugin should out live the engine.

Reply:
in_app_purchase may not be a problem in pure flutter projects, but in projects that natively access flutter, the engine cannot be effectively released, resulting in memory leaks, for example:

```objective-c
@interface AppDelegate : UIResponder <UIApplicationDelegate>
@property (nonatomic, strong) FlutterEngineGroup *engines;
@end
```
```objective-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
        self.engines = [[FlutterEngineGroup alloc] initWithName:@"" project:nil];
        return YES;
}
```
**Controller A**
```objective-c
- (IBAction)pushNext:(id)sender {
    NextViewController *vc = [[NextViewController alloc] initWithEntryPoint:nil];
    [self.navigationController pushViewController:vc animated:YES];
}
```

**Controller B**
```objective-c
// .h
@interface NextViewController : FlutterViewController
- (instancetype)initWithEntryPoint:(NSString *)point ;
@end
// .m 
- (instancetype)initWithEntryPoint:(NSString *)point {
    AppDelegate *delegate = ((AppDelegate *)([UIApplication sharedApplication].delegate));

    FlutterEngine *engine = [delegate.engines makeEngineWithEntrypoint:point libraryURI:nil];
    [GeneratedPluginRegistrant registerWithRegistry:engine];
    if (self = [super initWithEngine:engine nibName:nil bundle:nil]) {}
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.channel = [FlutterMethodChannel methodChannelWithName:@"normal" binaryMessenger:self.engine.binaryMessenger];
}
```
When I pop NextViewController , the flutter engine is not released, I found that in_app_purchase caused by debug memory graph.
The registry is not used, I don't know why it is written like this, I think it can be deleted.